### PR TITLE
Deprecate engcrk.xml

### PR DESCRIPTION
**Pull request by Madoshakalaka**
engcrk.xml now is extracted at runtime and no longer needed. Also with documentation cleanup.

### CODE REVIEW

File: CreeDictionary/DatabaseManager/english_keyword_extraction.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    """
+    returns lowercased keywords from the translation, by stripping stop words like a/an s/he s.o. s.t.
+
+    >>> extract_keywords("s/he is of a that kind (previously mentioned)") == {'is', 'of', 'that', 'kind', 'previously', 'mentioned'}
+    True
+    """
+    words = set(word_pattern.findall(translation.lower()))
+
+    # .strip('.') is because of the trailing periods in some translation
+    return {word.strip(".") for word in words if word not in stop_words}
```
Comment (Madoshakalaka) [inline CreeDictionary/DatabaseManager/english_keyword_extraction.py:19]: @aarppe this is the incarnation of your awk script, as you may wanted, the `stop_words` set is the same.

File: cypress/integration/regressions.spec.js [hunk 2]
```diff
@@ -40,11 +40,6 @@ context('Regressions', () => {
 
   // https://github.com/UAlbertaALTLab/cree-intelligent-dictionary/issues/158
   it('should display relevant English results', () => {
-    cy.visitSearch('see')
-    cy.get('[data-cy=search-results]')
-      .should('contain', 'wâpiw')
-      .and('contain', 'wâpahtam')
-      .and('contain', 'wâpamêw')
```
Comment (Madoshakalaka) [inline cypress/integration/regressions.spec.js:47]: @aarppe @eddieantonio wapiw has translation "s/he sees, s/he has vision CW", with the current runtime keyword extraction, you'll have to type in "sees" have the result.

The English stemming/lemmatization can be improved by an English FST

File: libexec/crkeng_stop_word_stat.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+ ('things', 1233),
+ ('with', 1228),
+ ('by', 1201),
+ ('or', 1136),
+ ('people', 1083),
+ ('on', 1064),
+ ('makes', 999),
+ ('e.g.', 867),
+ ('as', 859),
+ ('his/her', 712),
```
Comment (Madoshakalaka) [inline libexec/crkeng_stop_word_stat.py:57]: @eddieantonio @aarppe I wrote a script that does statistics about stop words. Currently only five are used. Do you think we can be less conservative and use more than five of them?

File: cypress/integration/regressions.spec.js (hunk 2 above)
Comment (Madoshakalaka) [reply cypress/integration/regressions.spec.js:47]: @aarppe sorry, after some thinking I realized this is really bad. I'll add a dumb English stemmer real quick, in the process of English keyword extraction

File: libexec/crkeng_stop_word_stat.py (hunk 3 above)
Comment (eddieantonio) [reply libexec/crkeng_stop_word_stat.py:57]: I'd prefer a search index with tf.idf or Okapi BM25 instead of stop words, as it takes into account these statistics automatically.

File: CreeDictionary/tests/API_tests/model_test.py [hunk 4]
```diff
@@ -102,12 +102,22 @@ def test_search_for_exact_lemma(lemma: Wordform):
...
+
+    # s = snowballstemmer.EnglishStemmer()
+
+    # >>> s.stemWord('story')
+    # 'stori'
+    # >>> s.stemWord('stories')
+    # 'stori'
+
     # This should match "âcimowin" and related words:
     search_results = Wordform.search("story")
```
Comment (Madoshakalaka) [inline CreeDictionary/tests/API_tests/model_test.py:122]: limitation of our stemmer

File: libexec/crkeng_stop_word_stat.py (hunk 3 above)
Comment (Madoshakalaka) [reply libexec/crkeng_stop_word_stat.py:57]: sounds cool!

File: cypress/integration/regressions.spec.js (hunk 2 above)
Comment (Madoshakalaka) [reply cypress/integration/regressions.spec.js]: @aarppe @eddieantonio now a stemmer is used and things goes well :)

File: CreeDictionary/DatabaseManager/english_keyword_extraction.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+import snowballstemmer
+
+word_pattern = re.compile(r"[\w.\-/']+")
+
+stop_words = {"a", "an", "the", "s/he", "s.o.", "s.t."}
+
+stemmer = snowballstemmer.EnglishStemmer()
+
+
+def extract_keywords(translation: str) -> List[str]:
```
Comment (eddieantonio) [inline CreeDictionary/DatabaseManager/english_keyword_extraction.py:13]: Why was this changed to `List`?

File: CreeDictionary/tests/API_tests/model_test.py (hunk 4 above)
Comment (eddieantonio) [reply CreeDictionary/tests/API_tests/model_test.py:122]: Wait, it doesn't match story?

File: cypress/integration/search.spec.js [hunk 6]
```diff
@@ -370,7 +371,7 @@ context('Searching', () => {
   describe('display of the header', function () {
     const lemma = 'nîmiw'
     const wordclassEmoji = '➡️' // the arrow is the most consistent thing, which means verb
-    const inflectionalCategory = 'VAI-1'
+    const inflectionalCategory = 'VAI-v'
```
[the remark this answers is not in the document]
Comment (aarppe) [reply cypress/integration/search.spec.js:374]: `VAI-v` is an old CW code which has been replaced in the latest CW version with `VAI-1/2/3`.

However, there are a few dictionary entries where I've also retained the old entry and the old wordclass code, so recognizing `VAI-v` is a useful fallback option, though on the longer term that old coding will no longer be used.

File: libexec/crkeng_stop_word_stat.py (hunk 3 above)
Comment (aarppe) [reply libexec/crkeng_stop_word_stat.py:57]: For future reference, the short stop word list was motivated by ruling out articles/determiners `a`, `an`, `the` and the most formulaic of CW conventions, i.e. `s/he`,` s.o.` and `s.t.`. The motivation was the users might have good reasons to search with common prepositions like `of` and `with`.

On the longer term, such elements which are grammatical in English and which do not always have a correspondent in Cree might be replaced with a "grammar entry", explaining how the grammatical function conveyed by the English preposition (or the like) can be (morpho-syntactically) communicated in Cree (for lack of a proper matching word).

**Comment by Madoshakalaka** [comment]
@aarppe @eddieantonio There's also a nice little behaviour change with this PR.

Spaces used to stop English search like this, simply no result will be presented based on the English:

# Old Behavior

Screenshot from 2020-08-19 15-57-28

It will now at least try and match each word:

# New Behavior

Screenshot from 2020-08-19 15-57-19

I imagine presenting _atim_ when somebody searches for "labrador dog" will make them somewhat happy, rather than presenting nothing.

**Comment by aarppe** [comment]
Concerning _stove_ above, we'd still want all of the below, as enumerated in #142, i.e.

- [ ] `stove` should match _stovepipe_
- [ ] `stove-pipe` should match _stovepipe_
- [ ] `stove pipe` should match _stovepipe_

**Comment by eddieantonio** [comment]
> Concerning _stove_ above, we'd still want all of the below, as enumerated in #142, i.e.
> 
> * [ ]  `stove` should match _stovepipe_
> * [ ]  `stove-pipe` should match _stovepipe_
> * [ ]  `stove pipe` should match _stovepipe_

This one is going to require more thought and consideration, so I'm making it a separate issue! #524

File: CreeDictionary/API/models.py [hunk 7]
```diff
@@ -534,11 +535,11 @@ def fetch_lemma_by_user_query(
         # todo: allow inflected forms to be searched through English. (requires database migration
         #  since now EnglishKeywords are bound to lemmas)
         english_results: Set[EnglishResult] = set()
-        if " " not in user_query:  # a whole word
+        for stemmed_keyword in stem_keywords(user_query):
 
             # this requires database to be changed as currently EnglishKeyword are associated with lemmas
             lemma_ids = EnglishKeyword.objects.filter(
-                text__iexact=user_query, **extra_constraints
+                text__iexact=stemmed_keyword, **extra_constraints
```
Comment (eddieantonio) [inline CreeDictionary/API/models.py:542]: This is great!
